pkg/expression/builtin_string.go

```diff
@@ -768,7 +772,7 @@ func (c *reverseFunctionClass) getFunction(ctx sessionctx.Context, args []Expres
 	bf.tp.SetFlen(args[0].GetType().GetFlen())
 	addBinFlag(bf.tp)
 	var sig builtinFunc
-	if types.IsBinaryStr(argTp) {
+	if types.IsBinaryStr(argTp) || bitColumnArgs {
```

Comment:
What about replacing `bitColumnArgs` with `types.IsTypeBit(argTp)` here?

Reply:
The `bit` type can appear not only in column, but also through expression right? For example, `cast(xxx as bit(37))`. We should also cover these cases in this PR, so using the `argTp` sounds simpler.

Reply:
updated. https://github.com/pingcap/tidb/pull/50146/commits/62c0cc332fe6ce34cceb1cb6327a25fb567897bd